cmake/ament_doxygen_generate.cmake

```diff
@@ -15,102 +15,146 @@
+#                      these must have been installed. See ament_doxygen_find_external() documentation
+#                      for further reference.
 # :type DEPENDENCIES: list of strings
 #
 # @public
 #
-function(ament_doxygen_generate)
+function(ament_doxygen_generate target_name)
   cmake_parse_arguments(
     args
-    ""
-    "CONFIG_OVERLAY;PROJECT_NAME;INPUT_DIRECTORY;OUTPUT_DIRECTORY"
+    "STANDALONE;NO_INSTALL"
+    "PROJECT_NAME;CONFIG_OVERLAY;CONFIG_DIRECTORY;INPUT_DIRECTORY;BUILD_DIRECTORY;INSTALL_DIRECTORY"
     "DEPENDENCIES" ${ARGN})
-  set(CONFIG_DIRECTORY "${ament_cmake_doxygen_DIR}/../resources")
 
-  if (NOT args_PROJECT_NAME)
+  if (NOT DEFINED args_PROJECT_NAME)
     set(args_PROJECT_NAME "${PROJECT_NAME}")
   endif()
-  set(PROJECT_NAME "${args_PROJECT_NAME}")
 
-  set(BUILD_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_doxygen/${PROJECT_NAME}")
-
-  if (NOT args_INPUT_DIRECTORY)
-    set(args_INPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
+  if (NOT DEFINED args_CONFIG_DIRECTORY)
+    set(args_CONFIG_DIRECTORY "ament_cmake_doxygen/${PROJECT_NAME}")
   endif()
-  set(INPUT_DIRECTORY "${args_INPUT_DIRECTORY}")
+  get_filename_component(args_CONFIG_DIRECTORY "${args_CONFIG_DIRECTORY}"
+    REALPATH BASE_DIR "${CMAKE_CURRENT_BINARY_DIR}"
+  )
 
-  if (NOT args_OUTPUT_DIRECTORY)
-    if (NOT DOXYGEN_OUTPUT_ROOT)
-      set(DOXYGEN_OUTPUT_ROOT "${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_doxygen")
-    endif()
-    set(args_OUTPUT_DIRECTORY "${DOXYGEN_OUTPUT_ROOT}/${PROJECT_NAME}")
+  if (NOT DEFINED args_BUILD_DIRECTORY)
+    set(args_BUILD_DIRECTORY "${args_CONFIG_DIRECTORY}/output")
   endif()
-  set(OUTPUT_DIRECTORY "${args_OUTPUT_DIRECTORY}")
+  get_filename_component(args_BUILD_DIRECTORY "${args_BUILD_DIRECTORY}"
+    REALPATH BASE_DIR "${CMAKE_CURRENT_BINARY_DIR}"
+  )
 
-  if (DOXYGEN_DOT_FOUND)
-    set(HAVE_DOT YES)
-    get_filename_component(DOT_PATH "${DOXYGEN_DOT_EXECUTABLE}" DIRECTORY)
+  if (NOT DEFINED args_INPUT_DIRECTORY)
+    set(args_INPUT_DIRECTORY ".")
   endif()
+  get_filename_component(args_INPUT_DIRECTORY "${args_INPUT_DIRECTORY}"
+    REALPATH BASE_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
+  )
 
-  foreach(dep ${args_DEPENDENCIES})
-    ament_index_has_resource(has_tag doxygen_tags "${dep}")
-    if(has_tag)
-      ament_index_get_resource(tag doxygen_tags "${dep}")
-      list(GET tag 0 tagfile)
-      list(GET tag 1 htmldir)
-      file(RELATIVE_PATH htmldir "${OUTPUT_DIRECTORY}/html" "${htmldir}")
-      set(TAGFILES "${TAGFILES} \"${tagfile}=${htmldir}\"")
-    else()
-      message(WARNING "A Doxygen tagfile for ${dep} could not be found, ignoring")
-    endif()
-  endforeach()
-  set(GENERATE_TAGFILE "${OUTPUT_DIRECTORY}/${PROJECT_NAME}.tag")
-
-  configure_file("${CONFIG_DIRECTORY}/Doxyfile.in" "${BUILD_DIRECTORY}/Doxyfile" @ONLY)
+  if (args_STANDALONE)
+    ament_doxygen_aggregate_external(${target_name}_dependencies
```

Comment:
Nit: Consider referencing this function at the top of the file in the docstring.

Reply:
Sounds good. Done in